buildSrc/subprojects/build/src/main/groovy/org/gradle/build/docs/UserGuideTransformTask.groovy

```diff
@@ -230,8 +230,14 @@ class UserGuideTransformTask extends DefaultTask {
                     exampleElement.appendChild(sourceFileTitle)
 
                     Element programListingElement = doc.createElement('programlisting')
-                    if (file.endsWith('.gradle') || file.endsWith('.groovy') || file.endsWith('.java')) {
+                    if (file.endsWith('.java')) {
                         programListingElement.setAttribute('language', 'java')
+                    } else if (file.endsWith('.gradle')) {
+                        programListingElement.setAttribute('language', 'gradle')
```

Comment:
❓ Why distinguish `groovy` and `gradle` languages?

I see that the `gradle-hl.xml` file contains an arbitrary list of keywords from Gradle model element names.
I'm not sure this is a good thing because then e.g. `allprojects` is highlighted as a keyword despite being a simple method. It is confusing to me, and another thing to maintain.

Reply:
Hmm, they're all methods, but the intent is to call out interesting declarations. 

I only have the "major" blocks declared and "task". Otherwise .gradle files stand out as rather uncolorful. I agree this could be much better if we had something that parsed the syntax tree. 

I can omit this if you think it's not helpful. WDYT?

Reply:
Doesn't Groovy highlighting work?

Reply:
After looking at the examples in this PR description I get what you mean.
The Kotlin sample has `override` keyword, adding some colors. The Groovy doesn't. I suppose this depends a lot on what code is displayed. One could argue that the extra highlighting of arbitrary methods is disturbing.  I'd be in favor of using simple Groovy and simple Kotlin highlighting.

Reply:
+1 on not trying to highlight specific keywords for now. It will just become a maintenance headache in the long run. I'd be in favor of using Groovy highlighting first.

Reply:
Is there any chance we can get the highlighting information from an external service instead? Then we would have to maintain less.

Reply:
Okay, I'll omit Gradle highlighting because it's really tough to do without a syntax parser.